src/services/mediaDevicePreferences.ts

```diff
@@ -105,26 +104,30 @@ function registerNewMediaDevice(device: MediaDeviceInfo, devicesList: MediaDevic
  * @param devices list of available devices
  * @param inputList list of registered audio/video devices in order of preference
  * @param inputId id of currently selected input
- * @return updated devices list (null, if it has not been changed)
+ * @return {MediaDeviceInfo[]|null} updated devices list (null, if it has not been changed)
  */
-function promoteMediaDevice(devices: MediaDeviceInfo[], inputList: MediaDeviceInfo[], inputId: string | null) {
-	const newInputList = inputList.slice()
+function promoteMediaDevice(devices: MediaDeviceInfo[], inputList: MediaDeviceInfo[], inputId: InputId): MediaDeviceInfo[] | null {
+	if (!inputId || inputId === DEFAULT_ID) {
+		return null
+	}
 
 	// Get the index of the first plugged device
-	const availableDevices = devices.map(device => device.deviceId).filter(id => id !== 'default')
-	const firstPluggedIndex = newInputList.findIndex(device => availableDevices.includes(device.deviceId))
-	const insertPosition = firstPluggedIndex === -1 ? newInputList.length : firstPluggedIndex
+	const availableDevices = devices.map(device => device.deviceId).filter(id => id !== DEFAULT_ID)
+	const firstPluggedIndex = inputList.findIndex(device => availableDevices.includes(device.deviceId))
+	const insertPosition = firstPluggedIndex === -1 ? inputList.length : firstPluggedIndex
 
 	// Get the index of the currently selected device
-	const currentDevicePosition = newInputList.findIndex(device => device.deviceId === inputId)
+	const currentDevicePosition = inputList.findIndex(device => device.deviceId === inputId)
 
 	if (currentDevicePosition === insertPosition) {
 		// preferences list is unchanged
 		return null
 	}
 
+	const newInputList = inputList.slice()
 	let deviceToPromote = null
-	if (currentDevicePosition === -1 && inputId !== 'default' && inputId !== null) {
+
+	if (currentDevicePosition === -1) {
 		// If device was not registered in preferences list, get it from devices list
 		deviceToPromote = devices.find(device => device.deviceId === inputId)
```

Comment:
devices and preferences list should be have the same length as populating the pref list loops over devices.

Reply:
prefs list could be longer, if you used some device before that

Reply:
Only if we keep `_populatePreferences` =)

Reply:
> Only if we keep _populatePreferences =)

Yes, it is kept, only browser storage part is skipped

Reply:
As written below, we'd miss new devices in the list in some cases, so I'd keep browser storage part